cpp/locality/NeighborList.cc

```diff
@@ -182,22 +197,29 @@ void NeighborList::copy(const NeighborList& other)
 void NeighborList::validate(unsigned int num_query_points, unsigned int num_points) const
 {
     if (num_query_points != m_num_query_points)
+    {
         throw std::runtime_error("NeighborList found inconsistent array sizes.");
+    }
     if (num_points != m_num_points)
+    {
         throw std::runtime_error("NeighborList found inconsistent array sizes.");
+    }
 }
 
 unsigned int NeighborList::bisection_search(unsigned int val, unsigned int left, unsigned int right) const
```

Comment:
It might be possible to write this to use `std::lower_bound`. https://en.cppreference.com/w/cpp/algorithm/lower_bound

Reply:
I think that's out of scope for this PR since it requires nonzero thought, but I'm not opposed to that in general.

Reply:
Issue for this was created #689